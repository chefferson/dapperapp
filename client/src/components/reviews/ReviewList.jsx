import React, { useEffect, useState } from 'react';
import { reviews } from '../../../mock-data/reviews.js';
import ReviewListEntry from './ReviewListEntry.jsx';

const ReviewList = (props) => {

  console.log(props);

  const defaultView = 2;
  const totalReviewLength = reviews.results.length;
  const [length, setLength] = useState(defaultView);

  const showMoreReviews = () => {
    if (totalReviewLength > 2) {
      if (length <= totalReviewLength) {
        return <button onClick={() => setLength(length + 2)}>More Reviews</button>;
      }
    }
    return null;
  };

  const filterReviewsByRating = () => {
    return reviews.results.filter(singleReview => singleReview.rating > 0)
      .slice(0, length).map((review, index) =>
        <ReviewListEntry
          review={review}
          key={review.review_id}
          index={index}
        />
      );
  };

  return (
    <>
      {filterReviewsByRating()}
      {showMoreReviews()}
    </>
  );
};

export default ReviewList;