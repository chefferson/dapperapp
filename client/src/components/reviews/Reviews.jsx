import React from 'react';
import withClickReporting from '../../helpers/withClickReporting.js';
import ReviewList from './ReviewList.jsx';
import RatingBreakdown from './RatingBreakdown.jsx';

const Reviews = (props) => {
  const RatingBreakdownWithClickReporting = withClickReporting(RatingBreakdown, 'RatingBreakdown');
  return (
    <div>
      <h2>RATINGS & REVIEWS</h2>
      <>
        <RatingBreakdownWithClickReporting data={props}/>
      </>
    </div>
  );
};

export default Reviews;