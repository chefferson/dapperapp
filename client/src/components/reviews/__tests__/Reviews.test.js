/**
 * @jest-environment jsdom
 */

import React from 'react';
import ReactDOM from 'react-dom';
import Reviews from '../Reviews.jsx';
import ReviewList from '../Reviews.jsx';
import { render } from '@testing-library/react';

it ('Reviews renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Reviews />, div);
});

it ('Review list renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ReviewList />, div);
});