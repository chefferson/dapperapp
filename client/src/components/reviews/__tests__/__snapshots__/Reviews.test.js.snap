// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductBreakdown matches snapshot 1`] = `
<div>
  <div>
    <div
      className="product-breakdown"
    >
      <div
        className="progress-box"
      >
        <div>
           
          Size
           
        </div>
        <div
          className="progress-bar"
        >
          <div
            style={
              Object {
                "textAlign": "right",
                "width": "60%",
              }
            }
          >
             
            <img
              alt="downArrow_57235"
              className={null}
              data-id={null}
              onClick={null}
              src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            />
             
          </div>
        </div>
        <div
          className="progress-filler-desc"
        >
          <label
            className="progress-desc"
          >
             
            Too small
          </label>
          <label
            className="progress-desc"
          >
             
            Perfect
          </label>
          <label
            className="progress-desc"
          >
             
            Too big
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="product-breakdown"
    >
      <div
        className="progress-box"
      >
        <div>
           
          Width
           
        </div>
        <div
          className="progress-bar"
        >
          <div
            style={
              Object {
                "textAlign": "right",
                "width": "63%",
              }
            }
          >
             
            <img
              alt="downArrow_57236"
              className={null}
              data-id={null}
              onClick={null}
              src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            />
             
          </div>
        </div>
        <div
          className="progress-filler-desc"
        >
          <label
            className="progress-desc"
          >
             
            Too small
          </label>
          <label
            className="progress-desc"
          >
             
            Perfect
          </label>
          <label
            className="progress-desc"
          >
             
            Too big
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="product-breakdown"
    >
      <div
        className="progress-box"
      >
        <div>
           
          Comfort
           
        </div>
        <div
          className="progress-bar"
        >
          <div
            style={
              Object {
                "textAlign": "right",
                "width": "63%",
              }
            }
          >
             
            <img
              alt="downArrow_57237"
              className={null}
              data-id={null}
              onClick={null}
              src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            />
             
          </div>
        </div>
        <div
          className="progress-filler-desc"
        >
          <label
            className="progress-desc"
          >
             
            Poor
          </label>
          <label
            className="progress-desc"
          >
             
            
          </label>
          <label
            className="progress-desc"
          >
             
            Great
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="product-breakdown"
    >
      <div
        className="progress-box"
      >
        <div>
           
          Quality
           
        </div>
        <div
          className="progress-bar"
        >
          <div
            style={
              Object {
                "textAlign": "right",
                "width": "66%",
              }
            }
          >
             
            <img
              alt="downArrow_57238"
              className={null}
              data-id={null}
              onClick={null}
              src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            />
             
          </div>
        </div>
        <div
          className="progress-filler-desc"
        >
          <label
            className="progress-desc"
          >
             
            Poor
          </label>
          <label
            className="progress-desc"
          >
             
            
          </label>
          <label
            className="progress-desc"
          >
             
            Great
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
